presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -109,7 +110,7 @@ public static JsonParser createJsonParser(JsonFactory factory, Slice json)
     {
         // Jackson tries to detect the character encoding automatically when using InputStream
         // so we pass an InputStreamReader instead.
-        return factory.createParser(new InputStreamReader(json.getInput()));
+        return factory.createParser(new InputStreamReader(json.getInput(), UTF_8));
```

Comment:
What about tweaking surefire so that we have fixed, non-UTF-9 encoding when running tests? Just like we do with the time zone?

Reply:
Static analysis is also a good idea, but it can catch only the APIs blacklisted. 

> it only helps if you have tests which would fail, and most tests just use ASCII. It'd be interesting to have a custom Charset which is broken for ASCII,

Like UTF-16?

Reply:
> Like UTF-16?
 
Hah, yes! Does the JVM work with that as the default charset?

Reply:
JVM can be started (so HelloWorld works) but JCE won't work:
```
[INFO] Running com.facebook.presto.tests.TestDistributedQueriesIndexed
[ERROR] Tests run: 39, Failures: 1, Errors: 0, Skipped: 38, Time elapsed: 46.633 s <<< FAILURE! - in com.facebook.presto.tests.TestDistributedQueriesIndexed
[ERROR] init(com.facebook.presto.tests.TestDistributedQueriesIndexed)  Time elapsed: 46.37 s  <<< FAILURE!
java.lang.ExceptionInInitializerError
	at com.facebook.presto.tests.TestDistributedQueriesIndexed.createQueryRunner(TestDistributedQueriesIndexed.java:40)
Caused by: java.lang.SecurityException: Framework jar verification can not be initialized
	at com.facebook.presto.tests.TestDistributedQueriesIndexed.createQueryRunner(TestDistributedQueriesIndexed.java:40)
Caused by: java.security.PrivilegedActionException
	at com.facebook.presto.tests.TestDistributedQueriesIndexed.createQueryRunner(TestDistributedQueriesIndexed.java:40)
Caused by: java.security.cert.CertificateException: Could not parse certificate: java.io.IOException: java.lang.IllegalArgumentException: Illegal base64 character -2
	at com.facebook.presto.tests.TestDistributedQueriesIndexed.createQueryRunner(TestDistributedQueriesIndexed.java:40)
Caused by: java.io.IOException: java.lang.IllegalArgumentException: Illegal base64 character -2
	at com.facebook.presto.tests.TestDistributedQueriesIndexed.createQueryRunner(TestDistributedQueriesIndexed.java:40)
Caused by: java.lang.IllegalArgumentException: Illegal base64 character -2
	at com.facebook.presto.tests.TestDistributedQueriesIndexed.createQueryRunner(TestDistributedQueriesIndexed.java:40)
```

Might be https://bugs.openjdk.java.net/browse/JDK-8187023 or something like that. I really like the comment https://bugs.openjdk.java.net/browse/JDK-8160267?focusedCommentId=13964784&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13964784 -- seems like they didn't do this yet.

Note, if you'd like to re-run the experiment, you need to set `-Dfile.encoding=UTF-16` using maven-surefire-plugin's `argLine`. With `systemPropertyVariables` it gets set too late, after `Charset.defaultEncoding()` is determined.

☹️